superset/daos/dataset.py

```diff
@@ -364,17 +364,6 @@ def create_metric(
     @staticmethod
     def bulk_delete(models: Optional[list[SqlaTable]], commit: bool = True) -> None:
         item_ids = [model.id for model in models] if models else []
-        # bulk delete, first delete related data
```

Comment:
I'm not sure if this was required because when you bulk delete a table (using the `delete()` method) you bypass the ORM and that's where the cascade logic resides, but given we've adopted the "shift left" mentality it's something of a moot point, i.e., the database will handle the deletion irrespective of how the records are deleted (SQLAlchemy ORM, raw connection, MySQL/PostgreSQL CLI, etc.).